Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,26 +68,29 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
+          if stanza == :artifacts
+            result = {}
+            value.each do |e|
+              result[e.class.dsl_key] = e.summarize
+            end
+            value = result
           end
 
-          if stanza == :artifacts
-            value = Hash[
-              value.map do |k, v|
-                v = v.map do |a|
-                  next a.to_a if a.respond_to?(:to_a)
-                  next a.to_h if a.respond_to?(:to_h)
-                  a
-                end
-
-                [k, v]
+          if artifact_name
+            result = nil
+            value.each do |k, v|
+              if k == artifact_name
+                result = v
+                break
               end
-            ]
+            end
+            value = result
+          end
```

Comment:
Since `value` is a hash at this point, you can just do 

```ruby
if artifact_name
  value = value[artifact_name]
end
```

Also, I think this should be inside the `if stanza == :artifacts` block.